palace/fem/coefficient.hpp

```diff
@@ -32,12 +35,34 @@ class BdrGridFunctionCoefficient
   //          the FET, FET.Elem1, and FET.Elem2 objects cannot be shared
   const mfem::ParMesh &mesh;
   mfem::FaceElementTransformations FET;
-  mfem::IsoparametricTransformation T1, T2, TF;
-  mfem::Vector CF;
+  mfem::IsoparametricTransformation T1, T2;
 
-  void GetBdrElementNeighborTransformations(mfem::ElementTransformation &T,
-                                            const mfem::IntegrationPoint &ip,
-                                            mfem::Vector *C1 = nullptr);
+  bool GetBdrElementNeighborTransformations(int i, const mfem::IntegrationPoint &ip)
+  {
+    // Get the element transformations neighboring the element, and optionally set the
+    // integration point too.
+    return GetBdrElementNeighborTransformations(i, mesh, FET, T1, T2, &ip);
+  }
+
+  static bool UseElem12(const mfem::FaceElementTransformations &FET,
+                        const MaterialOperator &mat_op)
+  {
+    // For interior faces with no way to distinguish on which side to evaluate quantities,
+    // evaluate on both (using the average or sum, depending on the application).
+    return (FET.Elem2 && mat_op.GetLightSpeedMin(FET.Elem2->Attribute) ==
+                             mat_op.GetLightSpeedMax(FET.Elem1->Attribute));
```

Comment:
Yeah this is old code and at the time I was trying to just find a good way to break ties for the rare case here. It seems like for any choice there will always be easy to construct cases where the `Min` is the same, or the `Max`, or the `Min` is the same as the other's `Max`.

Projecting onto the normal would work, though seems a bit excessively expensive (these 3x3 matrices are not optimized in MFEM, though something I have thought about is that they should be for basic 2x2 and 3x3 cases).

Hm...

Reply:
I agree with you, and think we should use `Max` since the refractive index is always <= 1. Done in [69ef14a](https://github.com/awslabs/palace/pull/240/commits/69ef14a2ae51aff5e2e53a62c380760e44184303).